Sources/MainUnit.pas

```diff
@@ -203,6 +208,77 @@ procedure TMainForm.AboutActionExecute(Sender: TObject);
   AboutBox.ShowModal;
 end;
 
+procedure TMainForm.actLoadPGNFileExecute(Sender: TObject);
+var
+  SList : TStringList;
+  i, R, j, row : Integer;
+  s, s2, pgntags, white, black : AnsiString;
+  RegexObj: TRegExpr;
+begin
+  //load pgn file
+  RegexObj := TRegExpr.Create();
+  RegexObj.Expression := '((((\[.*?\])\s*)+)(1\..+?(0-1|1-0|1/2-1/2)))';
```

Comment:
The game can be also unfinished and end with `*`. Also, it could be a good idea to handle some non-standard options for game ending. See https://github.com/alex65536/Chess256/blob/master/Sources/PGNParser.inc#L170-L183.